### ISSUE TITLE
Do not define _XOPEN_SOURCE and _POSIX_C_SOURCE, when using non-POSIX features

### ISSUE THREAD

**Issue by marcusva** [kcat/openal-soft#4]
`-D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700` probably may cause non-Linux builds to break due to missing type definitions and functions. The switches are meant (as far as I understand the standard) to force conformance to POSIX and XSI, so everything not covered is not supposed to work until explicitly told otherwise (see  http://pubs.opengroup.org/onlinepubs/007904975/functions/xsh_chap02_02.html, second section, 3rd bullet point).

Using both defines will cause FreeBSD to break building the OSS back-end due to missing `u_long`, `u_char` type definitions, since those are not covered by the standard.

```
/usr/bin/cc  -DAL_ALEXT_PROTOTYPES -DAL_BUILD_LIBRARY -D_GNU_SOURCE=1 -D_LARGEFILE_SOURCE -D_LARGE_FILES -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -Dopenal_EXPORTS -std=c99  -O2 -g -D_DEBUG -fPIC -I/home/marcus/devel/projects/extern/openal-soft/include -I/home/marcus/devel/projects/extern/openal-soft/build -I/home/marcus/devel/projects/extern/openal-soft/OpenAL32/Include -I/home/marcus/devel/projects/extern/openal-soft/Alc -I/usr/local/include    -Winline -Wall -Wextra -fvisibility=hidden -pthread -o CMakeFiles/openal.dir/Alc/backends/oss.c.o   -c /home/marcus/devel/projects/extern/openal-soft/Alc/backends/oss.c
In file included from /home/marcus/devel/projects/extern/openal-soft/Alc/backends/oss.c:41:
/usr/include/sys/soundcard.h:243:5: error: unknown type name 'u_long'; did you mean 'long'?
    u_long      play_rate;      /* sampling rate                        */
```

**Comment by dscharrer** [comment]
> POSIX has to be enabled for non-Windows builds, for pthreads and such

Not on FreeBSD - there everything works without the feature macros. In my experience they are only really needed for glibc (although there may well be other *nix platforms that also need them).

I have successfully (cross-)compiled 1.16.0 for FreeBSD with the following patch:

https://github.com/arx/ArxPackages/blob/master/files/crossdev/etc/portage/mine/cross-patches/media-libs/openal/00-nongnu.patch

Note that the header for `pthread_set_name_np` is found in `pthread_np.h`, not in `pthread.h`.

**Comment by kcat** [comment]
Can you check current master? I've tried adding a check for if _POSIX_C_SOURCE is needed (and removed the need for _XOPEN_SOURCE altogether).

**Comment by dscharrer** [comment]
master builds fine for me without any patches with my FreeBSD cross-compiler toolchain.

**Comment by kcat** [comment]
Great, then I suppose this is fixed with commit 4e6acfa0d549281efcbe8293d95b76a37deb370d. If someone's still having issues on FreeBSD, please open a new issue.
